test/Microsoft.AspNetCore.Routing.Tests/Tree/TreeRouterTest.cs

```diff
@@ -153,6 +153,97 @@ public class TreeRouterTest
         }
 
         [Theory]
+        [InlineData("a/{*path}", "/a")]
+        [InlineData("a/{*path}", "/a/")]
+        public async Task TreeRouter_RouteAsync_MatchesCatchAll_NullValue(
+            string template,
+            string requestPath)
+        {
+            // Arrange
+            var next = new Mock<IRouter>();
+            next
+                .Setup(r => r.RouteAsync(It.IsAny<RouteContext>()))
+                .Callback<RouteContext>(c => c.Handler = NullHandler)
+                .Returns(Task.FromResult(true))
+                .Verifiable();
+
+            var firstRoute = CreateMatchingEntry(next.Object, template, order: 0);
+            var matchingRoutes = new[] { firstRoute };
+            var linkGenerationEntries = Enumerable.Empty<TreeRouteLinkGenerationEntry>();
+            var attributeRoute = CreateAttributeRoute(next.Object, matchingRoutes, linkGenerationEntries);
+            var context = CreateRouteContext(requestPath);
+
+            // Act
+            await attributeRoute.RouteAsync(context);
+
+            // Assert
+            Assert.NotNull(context.Handler);
+            Assert.Null(context.RouteData.Values["path"]);
+        }
+
+        [Theory]
+        [InlineData("a/{*path}", "/a")]
+        [InlineData("a/{*path}", "/a/")]
+        public async Task TreeRouter_RouteAsync_MatchesCatchAll_NullValue_DoesNotTrounceExistingValue(
```

Comment:
nit: Isn't there a better word than trounce? Like replace, win, or something more common in english?